python/nav/web/report/views.py

```diff
@@ -503,6 +482,58 @@ def add_report_widget(request):
     return HttpResponse()
 
 
+def report_cache(key_items, query_dict):
+    """Report caching decorator.
+
+    Any exception that occurs while attempting to get or set cache items are
+    logged and subsequently ignored. We can still do our work, albeit slower.
+
+    :param key_items: A list of values that make up the request key
+
+    :param query_dict: The request query parameter dictionary; used to
+                       extract the "advanced search" parameters used in the
+                       request, so these can be included in the cache key
+    """
+    keys = ['report'] + list(key_items) + [_query_dict_hash(query_dict)]
+    cache_key = ':'.join(str(k) for k in keys)
```

Comment:
This has the potential to cause problems when mixing python2 and 3

Reply:
Actually, I'm not quite sure what Django's cache subsystem will accept as cache keys, but if `six.text_type` is acceptable to Django, I would use that.

Reply:
Doesn't much help that Django 1.11's doc mentions use of `str()` as well :-P

https://docs.djangoproject.com/en/1.11/topics/cache/#cache-key-transformation

Reply:
Not very easy to find anything about this in the docs, but the code and the tests provide:

https://github.com/django/django/commit/2679bc0304574edf5ded0c72889fb2e4d8cac911#diff-c4790c03f34c6a5b2929b8508141831f